pallets/circuit/src/lib.rs

```diff
@@ -375,53 +378,39 @@ pub mod pallet {
                     bidding_timeouts_at <= &frame_system::Pallet::<T>::block_number()
                 })
                 .map(|(xtx_id, _bidding_timeouts_at)| {
-                    let mut local_ctx = Self::setup(
-                        CircuitStatus::PendingBidding,
-                        &Self::account_id(),
-                        Some(xtx_id),
-                    )
-                        .unwrap();
-
-                    // Ensure Circuit::PendingBidding status
-                    if local_ctx.xtx.status != CircuitStatus::PendingBidding {
-                        Self::kill(&mut local_ctx, CircuitStatus::DroppedAtBidding);
-                        return
-                    }
-
-                    let current_step = local_ctx.xtx.steps_cnt.0;
-                    for mut fsx in local_ctx.full_side_effects[current_step as usize].iter_mut() {
-                        let sfx_id = fsx.generate_id::<SystemHashing<T>, T>(local_ctx.xtx_id);
-                        if let Some(best_sfx_bid) = <PendingSFXBids<T>>::get(xtx_id, sfx_id) {
-                            fsx.best_bid = Some(best_sfx_bid);
-                        } else {
-                            // error - some FSX don't have bids
-                            Self::kill(&mut local_ctx, CircuitStatus::DroppedAtBidding);
+                    Machine::<T>::compile_infallible(
```

Comment:
Can we move this to the machine maybe so it's coherent with the revert logic below?